server/src/infra/entities/asset-job-status.entity.ts

```diff
@@ -0,0 +1,15 @@
+import { Column, Entity, JoinColumn, OneToOne, PrimaryColumn } from 'typeorm';
+import { AssetEntity } from './asset.entity';
+
+@Entity('asset_job_status')
+export class AssetJobStatusEntity {
+  @OneToOne(() => AssetEntity, { onDelete: 'CASCADE', nullable: true })
```

Comment:
This is a primary column and cannot be nullable. Can you add on update cascade as well?

Reply:
Hmm, this construction matches [`SmartInfoEntity`](https://github.com/immich-app/immich/blob/7f2b072f077d9a4c254d30363f0c0babc0c25e08/server/src/infra/entities/smart-info.entity.ts#L6-L11) and [`ExifEntity`](https://github.com/immich-app/immich/blob/7f2b072f077d9a4c254d30363f0c0babc0c25e08/server/src/infra/entities/exif.entity.ts#L8-L13). Should this one be different?

Your points seem reasonable but I'm not familiar enough with `typeorm` to make a call either way.

Reply:
I think they may be incorrectly labeled. Your migration adds `assetId` as a non-nullable column and with a foreign key to the asset table. There is no way to have a job record without a parent asset and the migration reflects that. I think you should remove the nullable indicator here.

The opposite isn't necessarily true though. You _can_ have an Asset without a jobStatus record, so you could put `nullable: true` on the other side of the relation, but I'm not sure if `nullable` has much significance on these types of relations tbh.